modules/ingester/ingester.go

```diff
@@ -164,8 +167,41 @@ func (i *Ingester) markUnavailable() {
 	i.stopIncomingRequests()
 }
 
-// PushBytes implements tempopb.Pusher.PushBytes
+// PushBytes implements tempopb.Pusher.PushBytes. Traces pushed to this endpoint are expected to be in the formats
+//  defined by ./pkg/model/v1
+// This push function is extremely inefficient and is only provided as a migration path from the v1->v2 encodings
 func (i *Ingester) PushBytes(ctx context.Context, req *tempopb.PushBytesRequest) (*tempopb.PushResponse, error) {
+	var err error
+	v1Decoder, err := model.NewSegmentDecoder(v1.Encoding)
+	if err != nil {
+		return nil, err
+	}
+	v2Decoder, err := model.NewSegmentDecoder(v2.Encoding)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, t := range req.Traces {
+		trace, err := v1Decoder.PrepareForRead([][]byte{t.Slice})
+		if err != nil {
+			return nil, fmt.Errorf("error calling v1.PrepareForRead %w", err)
+		}
+
+		now := uint32(time.Now().Unix())
+		v2Slice, err := v2Decoder.PrepareForWrite(trace, now, now)
```

Comment:
Right. Since this is only called during migration I didn't want to add any more complexity here then necessary.

Unfortunately, it doesn't quite mimic past behavior. If you wrote a trace from a minute after it was actually generated then it's possible that now/now would exclude it incorrectly from a search. I'm unsure how much we want to invest in this though given how the tiny percentage of traces this will impact.